app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -604,13 +610,36 @@ private void pauseDownload() {
   private void handleInstallButtonClick() {
     view.getLifecycle()
         .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
-        .flatMap(create -> view.installAppClick()
+        .flatMap(create -> accountManager.accountStatus())
+        .observeOn(viewScheduler)
+        .flatMap(account -> view.installAppClick()
             .flatMapCompletable(action -> {
               Completable completable = null;
               switch (action) {
                 case INSTALL:
                 case UPDATE:
-                  completable = downloadApp(action);
+                  completable = appViewManager.loadAppViewViewModel()
+                      .observeOn(viewScheduler)
+                      .flatMapCompletable(
+                          appViewViewModel -> downloadApp(action).observeOn(viewScheduler)
+                              .doOnCompleted(() -> {
+                                appViewAnalytics.clickOnInstallButton(
+                                    appViewViewModel.getPackageName(),
+                                    appViewViewModel.getDeveloper()
+                                        .getName(), action.toString());
+                                if (account.isLoggedIn()
+                                    && appViewManager.shouldShowRecommendsPreviewDialog()) {
+                                  view.showRecommendsDialog();
+                                  appViewAnalytics.sendLoggedInRecommendAppDialogShowEvent(
+                                      appViewViewModel.getPackageName());
+                                } else if (!account.isLoggedIn()
+                                    && appViewManager.canShowNotLoggedInDialog()) {
+                                  appViewNavigator.navigateToNotLoggedInShareFragmentForResult(
+                                      appViewViewModel.getPackageName());
+                                  appViewAnalytics.sendNotLoggedInRecommendAppDialogShowEvent(
+                                      appViewViewModel.getPackageName());
```

Comment:
I would start extracting some of this logic out of here. This method is beginning to be too big and hard to read.
Maybe one thing that would be good would be to have one chain per action.

Reply:
We can not extract one chain per action because the listener in use is RxViewClicks and it only holds one reference per listener, so it would only listen to the last action.
To simplify the code, i extracted the code that shows the dialog (for both the logged in and not logged in cases).